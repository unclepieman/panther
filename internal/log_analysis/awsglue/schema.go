package awsglue

/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

// Infers Glue table column types from Go types, recursively descends types

import (
	"github.com/panther-labs/panther/internal/log_analysis/awsglue/glueschema"
)

type Column = glueschema.Column

var (
	// RuleMatchColumns are columns added by the rules engine
	RuleMatchColumns = []Column{
		{
			Name:    "p_rule_id",
			Type:    glueschema.TypeString,
			Comment: "Rule id",
		},
		{
			Name:    "p_alert_id",
			Type:    glueschema.TypeString,
			Comment: "Alert id",
		},
		{
			Name:    "p_alert_context",
			Type:    glueschema.TypeString,
			Comment: "Additional alert context",
		},
		{
			Name:    "p_alert_creation_time",
			Type:    glueschema.TypeTimestamp,
			Comment: "The time the alert was initially created (first match)",
		},
		{
			Name:    "p_alert_update_time",
			Type:    glueschema.TypeTimestamp,
			Comment: "The time the alert last updated (last match)",
		},
		{
			Name:    "p_rule_tags",
			Type:    glueschema.ArrayOf(glueschema.TypeString),
			Comment: "The tags of the rule that generated this alert",
		},
		{
			Name:    "p_rule_reports",
			Type:    glueschema.MapOf(glueschema.TypeString, glueschema.ArrayOf(glueschema.TypeString)),
			Comment: "The reporting tags of the rule that generated this alert",
		},
	}

	// RuleErrorColumns are columns added by the rules engine
	RuleErrorColumns = append(
		RuleMatchColumns,
		Column{
			Name:    "p_rule_error",
			Type:    glueschema.TypeString,
			Comment: "The rule error",
		},
	)
)
