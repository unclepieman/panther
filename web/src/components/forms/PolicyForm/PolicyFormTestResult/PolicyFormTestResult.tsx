/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { ComplianceStatusEnum, TestPolicyRecord } from 'Generated/schema';
import { Card, Flex, Box, Heading, Text } from 'pouncejs';
import StatusBadge from 'Components/badges/StatusBadge';

interface PolicyFormTestResultProps {
  testResult: TestPolicyRecord;
}

const PolicyFormTestResult: React.FC<PolicyFormTestResultProps> = ({ testResult }) => {
  const {
    functions: { policyFunction },
    passed,
    name,
  } = testResult;

  return (
    <Card p={4} as="article">
      <Flex align="flex-start" spacing={4}>
        <StatusBadge status={passed ? ComplianceStatusEnum.Pass : ComplianceStatusEnum.Fail} />
        <Box spacing={2}>
          <Heading as="h2" size="x-small" fontWeight="medium">
            {name}
          </Heading>
          {policyFunction.error && (
            <React.Fragment>
              <Box as="dt" color="navyblue-100">
                Policy Body
              </Box>
              <Text as="dd" color="red-200">
                {policyFunction.error.message}
              </Text>
            </React.Fragment>
          )}
        </Box>
      </Flex>
    </Card>
  );
};

export default PolicyFormTestResult;
