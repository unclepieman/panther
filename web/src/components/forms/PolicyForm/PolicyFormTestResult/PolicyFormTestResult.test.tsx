/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { buildTestPolicyRecord, render } from 'test-utils';
import { ComplianceStatusEnum } from 'Generated/schema';
import PolicyFormTestResult from './PolicyFormTestResult';

describe('PolicyFormTestResult', () => {
  it('shows the necessary information', () => {
    const testResult = buildTestPolicyRecord({ passed: true });

    const { getByText } = render(<PolicyFormTestResult testResult={testResult} />);
    expect(getByText(testResult.name)).toBeInTheDocument();
    expect(getByText(testResult.error.message)).toBeInTheDocument();
    expect(getByText(ComplianceStatusEnum.Pass)).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const testResult = buildTestPolicyRecord();

    const { container } = render(<PolicyFormTestResult testResult={testResult} />);
    expect(container).toMatchSnapshot();
  });
});
