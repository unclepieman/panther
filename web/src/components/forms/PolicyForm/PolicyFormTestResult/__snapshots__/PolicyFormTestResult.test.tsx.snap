// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyFormTestResult matches the snapshot 1`] = `
.panther-6 {
  background-color: #23344e;
  border-radius: 4px;
  padding: 16px;
}

.panther-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.panther-5>*:not(:last-child) {
  margin-right: 16px;
}



.panther-1 {
  font-weight: 500;
  font-size: 1.125rem;
}

.panther-2 {
  color: #86a3c3;
}

.panther-3 {
  color: #e85959;
}

.panther-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #e85959;
}

<div>
  <header
    id="main-header"
  />
  <article
    class="panther-6"
  >
    <div
      class="panther-5"
    >
      <div
        aria-atomic="true"
        class="panther-0"
        role="status"
      >
        FAIL
      </div>
      <div
        class="panther-4"
        spacing="2"
      >
        <h2
          class="panther-1"
        >
          Utah
        </h2>
        <dt
          class="panther-2"
        >
          Policy Body
        </dt>
        <dd
          class="panther-3"
        >
          deposit
        </dd>
      </div>
    </div>
  </article>
  <footer
    id="footer"
  />
</div>
`;
